### PR TITLE
Reduce dependency on vgic_dist_device

### PR DESCRIPTION
**Pull request by axel-h**
- Pass `vgic` to internal functions and not `vgic_dist_device` to reduce dependencies
- Pass dedicated parameters to the internal fault handlers.

The way I understand the code is, that `vgic` is the main object we work with internally. The `vgic_dist_device` seem something that describes the mapping. I'm not sure where this is really needed nowadays, maybe in a non-CAmkES environment there are usecases. But this is completely internal, so my longterm-goal is droping this completely and store the information in `vgic`, see top commit of https://github.com/Hensoldt-Cyber/seL4_projects_libs/commits/patch-axel-6a. I just have to check how well this works with the GICv3 support that https://github.com/seL4/seL4_projects_libs/pull/58 brings in.

### CODE REVIEW

**Comment by kent-mcleod** [comment]
My preference is that we hold back these clean-up PRs until after GICv3 emulation support is merged.

**Comment by axel-h** [comment]
> My preference is that we hold back these clean-up PRs until after GICv3 emulation support is merged.

If adding GICv3 support is urgent for you, then I will hold this back.

File: libsel4vm/src/arch/arm/vgic/vgic.c [hunk 1]
```diff
@@ -523,75 +523,80 @@ int handle_vgic_maintenance(vm_vcpu_t *vcpu, int idx)
...
+        if (!is_pending(vgic->dist, virq_data->virq, vcpu->vcpu_id)) {
             virq_ack(vcpu, virq_data);
         }
     } else {
         DDIST("enabled irq %d has no handle\n", irq);
     }
 }
 
-static void vgic_dist_disable_irq(struct vgic_dist_device *d, vm_vcpu_t *vcpu, int irq)
+static int vgic_dist_disable_irq(vgic_t *vgic, vm_vcpu_t *vcpu, int irq)
```
Comment (kent-mcleod) [inline libsel4vm/src/arch/arm/vgic/vgic.c:561]: This needs to be void now.

File: libsel4vm/src/arch/arm/vgic/vgic.c [hunk 2]
```diff
@@ -523,75 +523,80 @@ int handle_vgic_maintenance(vm_vcpu_t *vcpu, int idx)
...
+    assert(vgic);
+    assert(vgic->dist);
     DDIST("disabling gic distributor\n");
-    gic_dist->enable = 0;
+    vgic->dist->enable = 0;
     return 0;
 }
 
-static void vgic_dist_enable_irq(struct vgic_dist_device *d, vm_vcpu_t *vcpu, int irq)
+static int vgic_dist_enable_irq(vgic_t *vgic, vm_vcpu_t *vcpu, int irq)
```
Comment (kent-mcleod) [inline libsel4vm/src/arch/arm/vgic/vgic.c:544]: This also is now void

File: libsel4vm/src/arch/arm/vgic/vgic.c (hunk 1 above)
Comment (axel-h) [reply libsel4vm/src/arch/arm/vgic/vgic.c]: fixed

File: libsel4vm/src/arch/arm/vgic/vgic.c (hunk 2 above)
Comment (axel-h) [reply libsel4vm/src/arch/arm/vgic/vgic.c]: fixed

**Review by kent-mcleod — APPROVED** [review]
I'd like to merge this before https://github.com/seL4/seL4_projects_libs/pull/58

**Pull request merged**